README.md

```diff
@@ -1,108 +1,73 @@
 
-## Build environment
+Example:
+```Python
+from numba_dpcomp import njit
+import numpy as np
 
-```bash
-conda create -n dpcompenv python=3.7 numba=0.53 scipy pybind11 tbb tbb-devel cmake pytest
-```
-
-LLVM is required.
-
-For now only one of latest version of LLVM is supported, the sha of it commit can be found in llvm-sha.txt in ther root of repository.
+@njit
+def foo(a, b):
+    return a + b
 
-## Build
-
-For build LLVM and MLIR paths should be specified.
-Compatible LLVM build can be taken from https://github.com/IntelPython/mlir-llvm-recipe artifacts from Azure build.
-It also can be build manually. Instruction below suppose LLVM to be placed in mlir-llvm folder in CWD.
-
-### Windows
-
-LLVM build:
-```bash
-git clone https://github.com/IntelPython/mlir-llvm-recipe.git
-cmake --version
-set /p SHA=<mlir-llvm-recipe\\llvm-sha.txt
-pip install ninja
-git clone https://github.com/llvm/llvm-project.git
-cd llvm-project
-git checkout %SHA%
-"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvars64.bat"
-cmake ..\\llvm-project\\llvm -G "Ninja" -DLLVM_ENABLE_PROJECTS=mlir -DLLVM_BUILD_EXAMPLES=ON -DLLVM_TARGETS_TO_BUILD="host" -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_ASSERTIONS=ON -DCMAKE_INSTALL_PREFIX=..\\mlir-llvm
-ninja install
+result = foo(np.array([1,2,3]), np.array([4,5,6]))
+print(result)
 ```
 
-dpcomp build:
-```bash
-git clone https://github.com/IntelPython/dpcomp.git
-curl --output tbb.zip -L "https://github.com/oneapi-src/oneTBB/releases/download/v2021.1.1/oneapi-tbb-2021.1.1-win.zip"
-mkdir tbb
-tar -xf "tbb.zip" -C tbb --strip-components=1
-<path_to_miniconda>\\Scripts\\activate
-conda install numba=0.53 scipy pybind11 tbb=2021.1 pytest -c conda-forge
-cmake --version
-set LLVM_PATH=%cd%\\mlir-llvm
-set TBB_PATH=%cd%\\tbb
-cd dpcomp
-python setup.py develop
-```
+## Building and testing Python frontend
 
+You will need LLVM built from specific commit, found in `llvm-sha.txt`.
 
-### Linux/MacOS
+### Linux
 
-LLVM build:
-```bash
-git clone https://github.com/IntelPython/mlir-llvm-recipe.git
-export SHA=$(cat mlir-llvm-recipe/llvm-sha.txt)
+Building llvm
+```Bash
 git clone https://github.com/llvm/llvm-project.git
 cd llvm-project
 git checkout $SHA
-cmake --version
-cmake ../llvm-project/llvm -DLLVM_ENABLE_PROJECTS=mlir -DLLVM_BUILD_EXAMPLES=ON -DLLVM_TARGETS_TO_BUILD="host" -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_ASSERTIONS=ON -DLLVM_ENABLE_RTTI=ON -DCMAKE_INSTALL_PREFIX=../mlir-llvm
-make -j$(nproc --all) all
-cmake --install .
+cd ..
+mkdir llvm-build
+cd llvm-build
+cmake ../llvm-project/llvm -GNinja -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS=mlir -DLLVM_ENABLE_ASSERTIONS=ON -DLLVM_ENABLE_RTTI=ON -DLLVM_USE_LINKER=gold -DCMAKE_INSTALL_PREFIX=../llvm-install
+ninja install
 ```
 
-dpcomp build:
-```bash
-git clone https://github.com/IntelPython/dpcomp.git
-wget -O tbb.tgz "https://github.com/oneapi-src/oneTBB/releases/download/v2021.1.1/oneapi-tbb-2021.1.1-lin.tgz"
+Getting TBB
```

Comment:
Since this is about a python compiler anyway, you might consider suggesting a conda-evn/v-env instead. This would make sure that other dependent python packages are available, too. See the readme in refactor branch.

Reply:
It would make this section shorter and things are easier for users.

Reply:
You mean using `tbb-devel` from conda? I actually tried this in the past and it didn't work for me as `tbb-devel` didn't contained required cmake files for `find_package(TBB)` to succeed.

Reply:
Hm, it seems to work on refactor (or it was never tested).